datafusion/physical-plan/src/execution_plan.rs

```diff
@@ -431,6 +434,20 @@ pub trait ExecutionPlan: Debug + DisplayAs + Send + Sync {
     fn cardinality_effect(&self) -> CardinalityEffect {
         CardinalityEffect::Unknown
     }
+
+    /// Attempts to push down the given projection into the input of this `ExecutionPlan`.
```

Comment:
I agree it makes sense to expose ProjectionExec here in this API -- it is a very special operator. In theory it would be nice to avoid the circular reference (`ProjectionExec` is an `ExecutionPlan` that depends on `ProjetionExec`) but if it compiles I say we 🚢 🇮🇹